File: leetcode/0202-happy-number/solution.py
class Solution:
    def isHappy(self, n: int) -> bool:
        res = n
        while True:
            if n == 1:
                return True
            if res < 10 and res % 2 == 0:
                break
            res = 0
            while n > 0:
                res = res + ((n % 10) * (n % 10))
                n //= 10
            n = res
        return False

File: leetcode/0202-happy-number/solution.rs
impl Solution {
    pub fn is_happy(mut n: i32) -> bool {
        if n < 10 {
            return n == 1 || n == 7;
        }

        let mut sum_digits = 0;
        
        while n > 0 {
            let digit = n % 10;
            sum_digits += digit * digit;
            n /= 10;
        }
        Self::is_happy(sum_digits)
    }
}

